File: DI/DI-Dagger-Java/src/main/java/model/casa/Casa.java
package model.casa;

import javax.inject.Inject;
import java.util.UUID;

// @Singleton // Veremos si funciona...
public class Casa {
    private final String id = UUID.randomUUID().toString();

    private final Ventana ventana;
    private final Puerta puerta;

    @Inject
    public Casa() {
        ventana = new Ventana();
        puerta = new Puerta();
    }

    public void entrar() {
        puerta.abrir();
        puerta.cerrar();
    }

    public void ventilar() {
        ventana.abrir();
    }

    @Override
    public String toString() {
        return "Casa{" +
                "id='" + id + '\'' +
                ", ventana=" + ventana +
                ", puerta=" + puerta +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/casas/Casa.kt
package models.casas

import java.util.*
import javax.inject.Inject


data class Casa
@Inject constructor(
    private val ventana: Ventana,
    private val puerta: Puerta,
) {
    val id: String = UUID.randomUUID().toString()

    fun entrar() {
        puerta.abrir()
        puerta.cerrar()
    }

    fun ventilar() {
        ventana.abrir()
    }
}
